syntax = "proto3";

package discovery;

// Message representing the service details
message ServiceInfo {
    string service_name = 1; // The type of service (e.g., 'django', 'user_service')
    string ip_address = 2;   // The IP address of the service
}

// Response message for registering a service
message RegisterServiceResponse {
    bool success = 1;      // Indicates if the registration was successful
    string message = 2;    // Optional message for additional info
}

// gRPC service definition for service discovery
service ServiceDiscovery {
    // RPC to register a service
    rpc RegisterService (ServiceInfo) returns (RegisterServiceResponse);
}
